test/e2e/kubectl.go

```diff
@@ -111,40 +111,31 @@ var _ = Describe("kubectl", func() {
 })
 
 func validateGuestbookApp(c *client.Client) {
-	// Wait for frontend to serve content.
-	serving := false
-	for start := time.Now(); time.Since(start) < guestbookResponseTimeout; time.Sleep(5 * time.Second) {
-		entry, err := makeRequestToGuestbook(c, "get", "")
-		if err == nil && entry == `{"data": ""}` {
-			serving = true
-			break
-		}
-	}
-	if !serving {
-		Failf("Frontend service did not start serving content in %v seconds", guestbookResponseTimeout.Seconds())
+	Logf("Waiting for frontend to serve content.")
+	if !waitForGuestbookResponse(c, "get", "", `{"data": ""}`) {
+		Failf("Frontend service did not start serving content in %v seconds.", guestbookResponseTimeout.Seconds())
 	}
 
-	// Try to add a new entry to the guestbook.
-	added := false
-	for start := time.Now(); time.Since(start) < guestbookResponseTimeout; time.Sleep(5 * time.Second) {
-		result, err := makeRequestToGuestbook(c, "set", "TestEntry")
-		if err == nil && result == `{"message": "Updated"}` {
-			added = true
-			break
-		}
-	}
-	if !added {
-		Failf("Cannot added new entry in %v seconds", guestbookResponseTimeout.Seconds())
+	Logf("Trying to add a new entry to the guestbook.")
+	if !waitForGuestbookResponse(c, "set", "TestEntry", `{"message": "Updated"}`) {
+		Failf("Cannot added new entry in %v seconds.", guestbookResponseTimeout.Seconds())
```

Comment:
I don't think we want as long a timeout here and below as we use for the guestbook service to start up. Can we use 2 separate timeouts?
* guestbookStartupTimeout = 10min when initially waiting for the guestbook components to schedule and the frontend to serve traffic
* guestbookResponseTimeout = ~60s when waiting for responses from a running guestbook service.